backbone.js

```diff
@@ -1493,6 +1493,25 @@
   // Cached regex for stripping urls of hash.
   var pathStripper = /#.*$/;
 
+  // Cached regex for detecting any character that is invalid in an encoded URI.
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI
+  var invalidURIEncoded = /[^;,/?:@&=+$a-zA-Z0-9-_.!~*'()#%]/;
+
+  // Cached regex for detecting invalid percent-encoded octets. Will match any
+  // "%" that is not followed by a hexadecimal number.
+  var invalidPercentOctets = /%(?![a-f0-9]{2}|[A-F0-9]{2})/;
```

Comment:
If it's a valid percent-encoding someone will eventually use it and ask us why we don't accept it. :smile: